hicommonwealth/commonwealth
# C3 Notification Data

## Link to Issue
Closes: #6785
Closes: #4372 

## Description of Changes
- Migration to update `Notification.notification_data.chain` and `Notification.notification_data.chain_id` to `community_id`. Also update `Notification.notification_data.author_chain` to `author_community_id`.
- Updates client-side models + all server-side references for Notifications for the changes described in the migration

## Test Plan
Ensure notification redirects are correct:
- View all 3 user dashboards and click on a row in each
- Open notifications modal (top right bell icon) and click on various notifications

Ensure subscriptions are working:
- Subscribed/Unsubscribe from the notification settings page
- Subscribe/Unsubscribe directly on Threads
- Subscribe/Unsubscribe community (aka new thread) notifications (left-bottom on sidebar)

Test that new thread/comment notification work:
- On account 1 subscribe to thread notifications for a specific community (left-bottom sidebar)
- On account 2 create a thread in that notification
- Refresh on account 1 and ensure notification appears in top-right notifications modal
